engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Type.java

```diff
@@ -519,29 +519,66 @@ private boolean isNothing(Node lib) {
   private Map<String, Function> methods() {
     if (methods == null) {
       CompilerDirectives.transferToInterpreter();
-      var allMethods = new HashMap<String, Function>();
-      var defScope = definitionScope.asModuleScope();
-      var methodsFromThisScope = defScope.getMethodsForType(this);
-      if (methodsFromThisScope != null) {
-        methodsFromThisScope.forEach(
+      methods = getMethods(true);
+    }
+    return methods;
+  }
+
+  /**
+   * Returns methods (both instance and static) defined on this type, including the ones inherited
+   * from super types. Instance methods are defined on this type, static methods are defined on its
+   * {@link #getEigentype() eigen type}. The methods defined on this type are searched for inside
+   * the module scope where this type is defined, so if there are any other extension methods
+   * defined in other modules, they are not included in the result.
+   *
+   * @param includeStaticMethods If static methods, defined on eigen type, should be included in the
+   *     result.
+   * @return All static and instance methods defined on this type, including the ones inherited from
+   *     Any.
+   */
+  @TruffleBoundary
+  public Map<String, Function> getMethods(boolean includeStaticMethods) {
+    var ctx = EnsoContext.get(null);
+    var allMethods = new HashMap<String, Function>();
+    for (var type : allTypes(ctx)) {
+      var methodsOnThisType = type.methodsOnThisType(includeStaticMethods);
+      allMethods.putAll(methodsOnThisType);
```

Comment:
What if there are overrides? This code will likely put the `Any.to_text` into the map and not `My_Type.to_text`... it may not matter as `type.getMethods(true).keySet()` - e.g. only keys are used, but then this method should be changed to `Set<String>` and not provide incorrect `.values()`.

Reply:
Fixed in 80e80e413961229554c046df451c49475101f299 - methods with same name from subtypes are kept in the map and not replaced by the methods from supertypes. In other words, overridden methods are preferred.